tests/sub_port_interfaces/sub_ports_helpers.py

```diff
@@ -857,3 +958,73 @@ def update_dut_arp_table(duthost, ip):
         ip: IP address of directly connected interface
     """
     duthost.command("ping {} -c 3".format(ip), module_ignore_errors=True)
+
+
+def configure_ptf_nn_agent(ptfhost, ptfadapter, ifaces):
+    """
+    Add new interfaces to interfaces map of ptfadapter
+    Args:
+        ptfhost: PTF host object
+        ptfadapter: PTF adapter
+        ifaces: List of interface names
+    """
+    ifaces = [ifaces] if not isinstance(ifaces, list) else ifaces
+    last_iface_id = sorted(ptfhost.host.options['variable_manager'].extra_vars['ifaces_map'].keys())[-1]
+
+    for iface_id, iface in enumerate(ifaces, last_iface_id+1):
+        ptfhost.host.options['variable_manager'].extra_vars['ifaces_map'][iface_id] = iface
+        ptfadapter.ptf_port_set.append(iface_id)
+
+    restart_ptf_nn_agent(ptfhost)
+
+    ptfadapter.reinit()
+
+
+def cleanup_ptf_nn_agent(ptfhost, ptfadapter, ifaces):
+    """
+    Remove interfaces from interfaces map of ptfadapter
+    Args:
+        ptfhost: PTF host object
+        ptfadapter: PTF adapter
+        ifaces: List of interface names
+    """
+    ifaces = [ifaces] if not isinstance(ifaces, list) else ifaces
+    ifaces_map = ptfhost.host.options['variable_manager'].extra_vars['ifaces_map']
+    config_port_indices = {v: k for k, v in ifaces_map.items()}
+
+    for iface in ifaces:
```

Comment:
LINE 997 is redundant as `iface_index` is poped out the `iface_map` from LINE996

Reply:
Makes sense. Done!